client.go

```diff
@@ -959,8 +959,16 @@ func clientDoDeadline(req *Request, resp *Response, deadline time.Time, c client
 	// Without this 'hack' the load on slow host could exceed MaxConns*
 	// concurrent requests, since timed out requests on client side
 	// usually continue execution on the host.
+
+	cleanup := false
 	go func() {
-		ch <- c.Do(reqCopy, respCopy)
+		errDo := c.Do(reqCopy, respCopy)
+		if cleanup {
```

Comment:
Without `sync/atomic` this will result in a race condition where one goroutine is reading `cleanup` and the other is possibly at the same time writing to it (if the timeout occurs at the same time as `c.Do` returning).